notification-service/src/main/java/com/redhat/parodos/notification/controller/NotificationRecordController.java

```diff
@@ -59,53 +59,70 @@ public class NotificationRecordController {
 
 	private final NotificationRecordService notificationRecordService;
 
-	private final PagedResourcesAssembler<NotificationRecord> notificationRecordPagedResourcesAssembler;
-
 	private final SecurityUtil securityUtil;
 
-	private final NotificationAssemblerDTO notificationAssemblerDTO = new NotificationAssemblerDTO();
+	private final ModelMapper modelMapper;
 
-	public NotificationRecordController(NotificationRecordService notificationRecordService,
-			PagedResourcesAssembler<NotificationRecord> notificationRecordPagedResourcesAssembler,
-			SecurityUtil securityUtil) {
+	public NotificationRecordController(NotificationRecordService notificationRecordService, SecurityUtil securityUtil,
+			ModelMapper modelMapper) {
 		this.notificationRecordService = notificationRecordService;
-		this.notificationRecordPagedResourcesAssembler = notificationRecordPagedResourcesAssembler;
 		this.securityUtil = securityUtil;
+		this.modelMapper = modelMapper;
 	}
 
 	/**
 	 * Returns all notifications for a user with option to apply filter or search term.
 	 */
+	@io.swagger.v3.oas.annotations.Operation(summary = "Returns a list of notification records for the user")
+	@ApiResponses(
+			value = { @ApiResponse(responseCode = "200", description = "Successfully retrieved page of notifications"),
+					@ApiResponse(responseCode = "401", description = "Unauthorized"),
+					@ApiResponse(responseCode = "403", description = "Forbidden") })
 	@GetMapping
-	@ResponseStatus(HttpStatus.OK)
-	public PagedModel<NotificationRecordResponseDTO> getNotifications(@PageableDefault(size = 100) Pageable pageable,
+	public ResponseEntity<Page<NotificationRecordResponseDTO>> getNotifications(
+			@PageableDefault(size = 100) Pageable pageable,
 			@RequestParam(value = "state", required = false) State state,
 			@RequestParam(value = "searchTerm", required = false) String searchTerm) {
-		if (securityUtil == null || securityUtil.getUsername() == null) {
-			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "No username associated with this request");
-		}
-		Page<NotificationRecord> notificationsRecordPage = this.notificationRecordService
-				.getNotificationRecords(pageable, securityUtil.getUsername(), state, searchTerm);
-		return this.notificationRecordPagedResourcesAssembler.toModel(notificationsRecordPage,
-				this.notificationAssemblerDTO);
+
+		Page<NotificationRecord> notificationsPage = this.notificationRecordService.getNotificationRecords(pageable,
+				securityUtil.getUsername(), state, searchTerm);
+		Page<NotificationRecordResponseDTO> notificationsDtoPage = notificationsPage
+				.map(NotificationRecord -> modelMapper.map(NotificationRecord, NotificationRecordResponseDTO.class));
```

Comment:
Should we have a method on `NotificationRecordResponseDTO.fromNotificationRecord(NotificationRecord)` we moved to that way in the workflow-service.

Reply:
I saw that was the way used both for `workflowDefinition` and `project` list (using modelMapper)
If we want to change it, lets address this change for all modules in another PR
(Stop using modelMapper)

Reply:
Why need to be addressed in a different issue?

What I mentioned is to use something like this, that it's much easier to do:
https://github.com/parodos-dev/parodos/blob/main/workflow-service/src/main/java/com/redhat/parodos/workflow/definition/dto/WorkDefinitionResponseDTO.java#L92-L105

Reply:
Changed
Anyway, need to change adittional places in the code (remove modelMapper usage)